### PR TITLE
Isa fixes

### PR DESCRIPTION
**Pull request by donofrio**
All RV64I instructions validated as well as many RV32I instructions

### CODE REVIEW

File: include/insns/RV32I.h [hunk 1]
```diff
@@ -37,7 +37,7 @@ namespace SST{
           }
           return true;
         }
-
+        Inst.imm = (Inst.imm & 0b011111111)*4;
         return addi(F,R,M,Inst);
       }
```
Comment (donofrio) [inline include/insns/RV32I.h:42]: we need to scale the immd by 4 per the spec

File: include/insns/RV32I.h [hunk 2]
```diff
@@ -168,6 +168,7 @@ namespace SST{
                        RevMem *M, RevInst Inst) {
         if( Inst.rd == 2 ){
           // c.addi16sp
+           SEXT(Inst.imm, (Inst.imm & 0b011111111)*16, 32)
           return addi(F,R,M,Inst);
         }else{
           // c.lui %rd, $imm = addi %rd, x0, $imm
```
Comment (donofrio) [inline include/insns/RV32I.h:174]: In this case it is scaled and sign extended per the spec for addi16sp

File: include/insns/RV64I.h [hunk 3]
```diff
@@ -115,13 +115,18 @@ namespace SST{
       }
 
       static bool addiw(RevFeature *F, RevRegFile *R,RevMem *M,RevInst Inst) {
-        R->RV64[Inst.rd] = dt_u64((int32_t)(td_u64(R->RV64[Inst.rs1],64)) + (int32_t)(td_u32(Inst.imm,12)),64);
+        R->RV64[Inst.rd] = dt_u32((int32_t)(td_u32(R->RV64[Inst.rs1],32)) + (int32_t)(td_u32(Inst.imm,12)),32);
         R->RV64[Inst.rd] &= MASK32;
         SEXTI( R->RV64[Inst.rd], 32 );
         R->RV64_PC += Inst.instSize;
         return true;
       }
```
Comment (donofrio) [inline include/insns/RV64I.h:124]: Addiw operates on 32-bit values

File: include/insns/RV64I.h [hunk 4]
```diff
@@ -115,13 +115,18 @@ namespace SST{
...
 
+   //     R->RV64[Inst.rd] = dt_u32(td_u32(R->RV64[Inst.rs1],32) + td_u32(R->RV64[Inst.rs2],32),32); //addw operates on and produces 32-bit results even on RV64I codes 
+   //     SEXTI(R->RV64[Inst.rd], 32);    //Sign extend the result up to 64bits
+   //     R->RV64_PC += Inst.instSize;
+   //     return true;
+
```
Comment (donofrio) [inline include/insns/RV64I.h]: sorry these snuck in here... are removed in a later commit

File: include/insns/RV32I.h [hunk 5]
```diff
@@ -588,7 +590,7 @@ namespace SST{
           }
           R->RV32_PC += Inst.instSize;
         }else{
-          if( R->RV64[Inst.rs1] < tmp ){
+          if( (R->RV64[Inst.rs1]) < tmp64 ){
             R->RV64[Inst.rd] = 1;
           }else{
             R->RV64[Inst.rd] = 0;
```
Comment (donofrio) [inline include/insns/RV32I.h:596]: There was some auto-casting happening in the background that prevented this from operating as an unsigned comparison, making it explicitly uint64_t addressed that

File: CMakeLists.txt [hunk 6]
```diff
@@ -30,7 +30,7 @@ option(CTEST_MULTILIB_TESTS "Enable 32-bit tests" ON)
 #---------------------------------------------------------------------------
 # Setup the path to the SST installation directory
 #set(SST_INSTALL_DIR "" CACHE PATH "SST installation directory")
-execute_process(COMMAND sst-config --prefix 
+execute_process(COMMAND sst-config --prefix
                 OUTPUT_VARIABLE SST_INSTALL_DIR
                 OUTPUT_STRIP_TRAILING_WHITESPACE
```
Comment (jleidel) [inline CMakeLists.txt:35]: good catch!

**Review by jleidel — APPROVED** [review]
looks solid

**Pull request merged**